Refactor `wcadmin_storeprofiler_store_business_features_continue`  Tracks event handling

octaedro opened the pull request:
Fixes #7912

This PR refactors the `wcadmin_storeprofiler_store_business_features_continue` Tracks event handling to not hardcode the extensions' names..

This Tracks event will save the props `install_[extension name]` with a boolean value.

The prop `install_wcpay ` has been renamed for `install_woocommerce_payments`. //cc. @pmcpinto

No changelog required.

### Screenshots

![screenshot-one wordpress test-2021 11 07-17_11_45](https://user-images.githubusercontent.com/1314156/140660492-fbb1f0a6-53cc-4135-8a95-788850902bb8.png)

### Detailed test instructions:

1. Open the browser devtools, go to the `Console` and enable the debug messages, you can do this by following the steps detailed here(P90Yrv-1Wj-p2 #debug-devtools).
2. Go to 4th step of the Onboarding wizard (Business Details) and click `Free features`.
3. Verify that the event `wcadmin_storeprofiler_store_business_features_continue` is recorded with the selected extensions props (e.g.: `install_google_listings_and_ads`, `install_mailpoet`) after pressing `Continue`.

<!--
Please add your test instructions to `/TESTING-INSTRUCTIONS.md`.
-->

<!--- Please add a Changelog note

Enter a changelog note using the following CLI command `npm run changelogger -- add` and commit changes. --->

octaedro committed:
Refactor tracks event handling

Refactor `wcadmin_storeprofiler_store_business_features_continue` tracks event handling to not hardcode extensions

octaedro committed:
Rename const and move function

octaedro committed:
Add tests

octaedro committed:
Fix typo

joshuatf commented on client/profile-wizard/steps/business-details/flows/selective-bundle/index.js:
```diff
@@ -58,6 +58,26 @@ export const filterBusinessExtensions = ( extensionInstallationOptions ) => {
 	);
 };
 
+export const prepareInstalledExtensionsForTracking = (
```
Very nitpicky, but for brevity and since this applies to both installed and not installed extensions:

```suggestion
export const prepareExtensionTrackingData = (
```

octaedro replied:
Sounds good to me. The method was renamed in commit 1dc80b9

joshuatf commented on client/profile-wizard/steps/business-details/test/index.js:
```diff
@@ -33,6 +36,71 @@ describe( 'BusinessDetails', () => {
+
+			expect( installedExtensions ).toEqual( expectedExtensions );
+		} );
+		test( 'preparing shipping and tax extensions for tracking', () => {
```
❤️  Awesome test

joshuatf reviewed (approved):
Great refactor, Fernando!  Left one nitpicky naming comment, but this is testing well. 🎉 

Pre-approving, but would like @pmcpinto's buy-in for the tracks data change on WC Pay before merge.

octaedro committed:
Renamed method

This commit renames the method `prepareInstalledExtensionsForTracking` as `prepareExtensionTrackingData`.

octaedro committed:
Prop renamed

octaedro commented:
Thank you @joshuatf for your review.
I addressed the suggested changes. Could you give this PR another look?

joshuatf commented on client/profile-wizard/steps/business-details/flows/selective-bundle/index.js:
```diff
@@ -58,6 +58,27 @@ export const filterBusinessExtensions = ( extensionInstallationOptions ) => {
+	) ) {
+		const key =
+			fieldKey === 'woocommerce-payments'
+				? 'install_wcpay'
```
👍

joshuatf reviewed (approved):
Testing well and the `install_wcpay` prop stays consistent.  LGTM!
